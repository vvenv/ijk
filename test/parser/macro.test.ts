import { expect, test } from 'vitest';
import { parse } from './__helper';

test('basic', () => {
  expect(
    parse(`{{ macro name x y }}{{x}}{{y}}{{ endmacro }}{{ name(1, 2) }}`),
  ).toMatchSnapshot();
});

test('default params', () => {
  expect(
    parse(`{{ macro name x="foo" y=123 }}{{x}}{{y}}{{ endmacro }}{{ name() }}`),
  ).toMatchSnapshot();
});

test.skip('call', () => {
  expect(
    parse(
      `{{ macro name foo bar }}{{foo}}{{bar}}{{ endmacro }}{{ call name(1, 2) }}3{{ endcall }}`,
    ),
  ).toMatchSnapshot();
});
