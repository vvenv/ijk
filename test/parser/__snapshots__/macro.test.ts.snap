// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic 1`] = `
{
  "out": Out {
    "content": ""use strict";let s = "";const _m_1_0 = (x,y)=>{const c_m_1_0={...c,x,y,};s+=e(c_m_1_0.x);s+=e(c_m_1_0.y);return "";};c.name = _m_1_0;s+=e(c.name(1, 2));return s;",
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": true,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
  },
  "smp": SMP {
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": true,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
    "template": "{{ macro name x y }}{{x}}{{y}}{{ endmacro }}{{ name(1, 2) }}",
  },
}
`;

exports[`default params 1`] = `
{
  "out": Out {
    "content": ""use strict";let s = "";const _m_1_0 = (x="foo",y=123)=>{const c_m_1_0={...c,x,y,};s+=e(c_m_1_0.x);s+=e(c_m_1_0.y);return "";};c.name = _m_1_0;s+=e(c.name());return s;",
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": true,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
  },
  "smp": SMP {
    "options": {
      "autoEscape": true,
      "debug": false,
      "strictMode": true,
      "stripComments": true,
      "tagEnd": "}}",
      "tagStart": "{{",
      "trimWhitespace": true,
    },
    "template": "{{ macro name x="foo" y=123 }}{{x}}{{y}}{{ endmacro }}{{ name() }}",
  },
}
`;
